### PR TITLE
Feat | Retrieve message API

### PR DESCRIPTION
**Pull request by APIYOJENNIFER**
With reference to issue #3, this PR will enable authenticated users to retrieve messages posted in the groups they belong in

### CODE REVIEW

File: group/urls.py [hunk 1]
```diff
@@ -8,4 +8,5 @@
     path('detail/<int:group_id>/', GroupDetailApiView.as_view()),
     path('user/<int:user_id>/', GroupDetailApiView.as_view()),
     path('message', MessageAPIView.as_view()),
+    path('messages', MessageAPIView.as_view()),
```
Comment (ThatcherK) [inline group/urls.py:14]: I would expect this url to be the same as the one on line 10, with the difference being a primary key identifier, say for the user making the get request for their messages . i.e message/<user_id>
Please refer to this resource for RESTFUL API best practices https://www.sitepoint.com/build-restful-apis-best-practices/

**Review by ThatcherK — CHANGES_REQUESTED** [review]

**Review by ThatcherK — APPROVED** [review]

File: group/views.py [hunk 2]
```diff
@@ -182,3 +182,26 @@ def post(self, request):
...
         return Response({"error": "User is not a member of this group"},
                         status=status.HTTP_403_FORBIDDEN)
+
+    def get(self, request, user_id):
+        """Retrieve all messages in a group"""
+        data = request.data
+        group_id = data.get("group_id")
+        try:
+            group = Group.objects.get(id=group_id)
+            user = User.objects.get(id=user_id)
```
Comment (wakibi) [inline group/views.py]: Use `select_related` here. That should also take care of the if condition on line 201.
Propose factoring the post method as well to use the same idea

File: group/urls.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
     path('', GroupApiView.as_view()),
     path('<int:group_id>/', GroupApiView.as_view()),
     path('detail/<int:group_id>/', GroupDetailApiView.as_view()),
-    path('user/<int:user_id>/', GroupDetailApiView.as_view()),
-    path('message', MessageAPIView.as_view()),
+    path('users/<int:user_id>/', GroupDetailApiView.as_view()),
+    path('messages', MessageAPIView.as_view()),
+    path('messages/<int:user_id>/', MessageAPIView.as_view()),
```
Comment (wakibi) [inline group/urls.py]: Why are we using the same class on this URL as the one above it?

File: group/views.py [hunk 4]
```diff
@@ -182,3 +182,26 @@ def post(self, request):
                             status=status.HTTP_400_BAD_REQUEST)
         return Response({"error": "User is not a member of this group"},
                         status=status.HTTP_403_FORBIDDEN)
+
+    def get(self, request, user_id):
```
Comment (wakibi) [inline group/views.py]: Suggest adding a default on the `user_id` here if you really want to use the same class for both URLs

**Review by wakibi — CHANGES_REQUESTED** [review]

File: group/urls.py (hunk 3 above)
Comment (APIYOJENNIFER) [reply group/urls.py]: If you are referring to the MessageAPIView class then it only makes sense for both the POST and GET methods to be within the same class hence the URLs

**Comment by APIYOJENNIFER** [comment]
@wakibi could you please take a look at this when you get time

File: group/urls.py (hunk 3 above)
Comment (wakibi) [reply group/urls.py]: I took a second look at this, and it's fundamentally wrong by design. We shall have a quick checkin tomorrow to show you why.

**Review by wakibi — APPROVED** [review]

**Pull request merged**